operator/pkg/secretsync/cell.go

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package secretsync
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	ctrlRuntime "sigs.k8s.io/controller-runtime"
+
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+)
+
+// Cell manages K8s Secret synchronization from application namespaces
+// into dedicated Cilium secrets namespace.
+//
+// Subsystems that are interested in having K8s Secrets synced
+// (e.g. Gateway API, Ingress, ...) can register themselves via
+// SecretSyncRegistrationOut.
+//
+// This way, multiple use-cases are sharing the same reconciler.
+// This potentially prevents multiple reconcilers from interfering
+// with each other.
+//
+// Example:
+//
+// cell.Provide(func registerSecretSyncRegistration(...) secretsync.SecretSyncRegistrationOut {...})
+var Cell = cell.Module(
+	"secret-sync",
+	"Syncs TLS secrets into a dedicated secrets namespace",
+
+	cell.Invoke(initSecretSyncReconciliation),
+)
+
+type secretSyncParams struct {
+	cell.In
+
+	Logger    logrus.FieldLogger
+	Lifecycle hive.Lifecycle
+
+	CtrlRuntimeManager ctrlRuntime.Manager
+	Registrations      []*SecretSyncRegistration `group:"secretSyncRegistrations"`
```

Comment:
I thought about the dependency with an explicit registration too.

But this would come with the drawback of having to register the secret-sync reconciler with the controller-runtime manager in a lifecycle hook (instead of in the cunstructor). That's because we only have all the information AFTER all the cells would have registered their need for a secret-sync. And this would potentially result in error handling after the leader election of the operator.

By using the `group` all registrations are already known.

That's why i went with this approach.

Reply:
Ah, I see- imho, if there's a reason for using group that's fine by me - it's not like I don't want it to be used at all. Pop the above in a comment somewhere and we're all good :)